calculator_test.go

```diff
@@ -7,162 +7,176 @@ import (
-		want := tc.want
+	for _, tc := range testCases {
 		got := calculator.Add(tc.a, tc.b)
-		if want != got {
-			t.Errorf("test %s: want %f, got %f", tc.comment, want, got)
+		if tc.want != got {
+			t.Errorf("test %s: want %f, got %f", tc.comment, tc.want, got)
 		}
 	}
 }
 
 func TestAddRandom(t *testing.T) {
 	t.Parallel()
 	for i := 0; i < 100; i++ {
-		a := rand.Float64()
-		b := rand.Float64()
+		a := rand.Float64() + 1
+		b := rand.Float64() + 1
 		want := a + b
 		got := calculator.Add(a, b)
 		if want != got {
-			t.Errorf("random test: want %f, got %f", want, got)
+			t.Errorf("Add(%f, %f): want %f, got %f", a, b, want, got)
 		}
 	}
 }
 
 func TestSubtract(t *testing.T) {
+	testCases := []struct {
+		a, b    float64
+		want    float64
+		comment string
+	}{
+		{a: 1, b: 2, want: -1, comment: "one minus two"},
+		{a: 2, b: 5, want: -3, comment: "two minus five"},
+	}
 	t.Parallel()
-	for _, tc := range testCasesSubtract {
-		want := tc.want
+	for _, tc := range testCases {
 		got := calculator.Subtract(tc.a, tc.b)
-		if want != got {
-			t.Errorf("test %s: want %f, got %f", tc.comment, want, got)
+		if tc.want != got {
+			t.Errorf("test %s: want %f, got %f", tc.comment, tc.want, got)
 		}
 	}
 }
 
 func TestSubtractRandom(t *testing.T) {
 	t.Parallel()
 	for i := 0; i < 100; i++ {
-		a := rand.Float64()
-		b := rand.Float64()
+		a := rand.Float64() + 1
+		b := rand.Float64() + 1
 		want := a - b
 		got := calculator.Subtract(a, b)
 		if want != got {
-			t.Errorf("random test: want %f, got %f", want, got)
+			t.Errorf("Subtract(%f, %f): want %f, got %f", a, b, want, got)
 		}
 	}
 }
 
 func TestMultiply(t *testing.T) {
+	testCases := []struct {
+		a, b    float64
+		want    float64
+		comment string
+	}{
+		{a: 1, b: 2, want: 2, comment: "one times two"},
+		{a: 2, b: 5, want: 10, comment: "two times five"},
+	}
 	t.Parallel()
-	for _, tc := range testCasesMultiply {
-		want := tc.want
+	for _, tc := range testCases {
 		got := calculator.Multiply(tc.a, tc.b)
-		if want != got {
-			t.Errorf("test %s: want %f, got %f", tc.comment, want, got)
+		if tc.want != got {
+			t.Errorf("test %s: want %f, got %f", tc.comment, tc.want, got)
 		}
 	}
 }
 
 func TestMultiplyRandom(t *testing.T) {
 	t.Parallel()
 	for i := 0; i < 100; i++ {
-		a := rand.Float64()
-		b := rand.Float64()
+		a := rand.Float64() + 1
+		b := rand.Float64() + 1
 		want := a * b
 		got := calculator.Multiply(a, b)
 		if want != got {
-			t.Errorf("random test: want %f, got %f", want, got)
+			t.Errorf("Multiply(%f, %f): want %f, got %f", a, b, want, got)
 		}
 	}
 }
 
 func TestDivide(t *testing.T) {
+	testCases := []struct {
+		a, b        float64
+		want        float64
+		comment     string
+		errExpected bool
+	}{
+		{a: 2, b: 1, want: 2, errExpected: false, comment: "two divided by 1"},
+		{a: 10, b: 5, want: 2, errExpected: false, comment: "ten divided by 5"},
+		{a: 10, b: 0, want: 0, errExpected: true, comment: "error out on divide by zero"},
```

Comment:
In the 'error expected' cases, try using completely ridiculous `want` values, such as `1_000_000`, or `999`. That way, if you accidentally compare the data value when you shouldn't, you'll see that reflected in the output.

Reply:
sorted, thanks man